File: src/main/java/com/imp/checklist/EmptyChecklist.java
package com.imp.checklist;

import com.imp.checklist.question.Question;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.ArrayList;

@Getter
@Setter
@Document("checklists")
public class EmptyChecklist implements Checklist{
    @Id
    private String id;
    private String creatorId;
    private ArrayList<String> artifactTypes;
    private ArrayList<Question> questions;
}

File: src/main/kotlin/com/julia/imp/checklist/EmptyChecklist.kt
package com.julia.imp.checklist

import com.julia.imp.question.Question
import org.bson.codecs.pojo.annotations.BsonId
import org.bson.types.ObjectId

data class EmptyChecklist (
    @BsonId
    val id: ObjectId,
    val creatorId: String,
    val artifactTypes: ArrayList<String>,
    val questions: ArrayList<Question>
)
